### DISCUSSION TITLE
Number of simultaneous devices

### DISCUSSION THREAD

**Discussion by BeckmaR** [energomonitor/bisquitt#25] [Q&A] [answered]
Dear team,

I discovered bisquitt today. Did you ever execute any tests regarding the maximum amount of devices that is feasible to connect at onceusing the bisquitt + mosquitto setup? Are there any experiences?

**Comment by mprymek** [comment, marked as answer]
Hi!

Thank you for your interest. We have a stresser tool (in-house, unpublished) and performed many tests some time ago. Unfortunately, they were in-house tests and we don't have any article or publication summarising the results.

What I can say for sure, is that the performance is very good but concrete figures are highly hardware-dependant. The number of clients per se is virtually unlimited. The only practical limit is the number of messages per second and their size. As the underlying UDP protocol has no flow control, if the hardware is not fast enough to process the packets, the network buffers overflow and you start getting lost packets, MQTT-SN timeouts and resends and situation is getting worse... Where exactly the limit is, depends highly on the system CPU, network throughput, OS,...

So I really must recommend to do performance tests on the concrete hardware and with a traffic as much similar to your production traffic as possible. We could probably provide some help to you, if it's needed. You can contact me on the email provided here in my GH account.
